var/spack/repos/builtin/packages/py-jupyter-notebook/package.py

```diff
@@ -37,3 +39,8 @@ class PyJupyterNotebook(PythonPackage):
     depends_on('py-ipykernel@5.1.0:', when='@4.2.0:', type=('build', 'run'))
     depends_on('py-terminado@0.3.3:', when="+terminal", type=('build', 'run'))
     depends_on('py-ipywidgets', when="+terminal", type=('build', 'run'))
+    # required for 6.0.1, not sure about 5.x.x, as there is no version in spack
+    # right now treat them as 6-or-newer dependencies
+    depends_on('py-prometheus-client', type=('build', 'run'), when='@6:')
+    depends_on('py-terminado', type=('build', 'run'), when='@6:')
+    depends_on('py-send2trash', type=('build', 'run'), when='@6:')
```

Comment:
Missing dependencies and version constraints on:
```python
        'pyzmq>=17',                                                            
        'traitlets>=4.2.1',                                                     
        'jupyter_core>=4.4.0',                                                  
        'jupyter_client>=5.3.1',    
        'terminado>=0.8.1',                                                     
        ':python_version == "2.7"': ['ipaddress'],                              
```
Also, `py-jupyter-console` is not required.